include/range/v3/utility/iterator.hpp

```diff
@@ -480,25 +523,111 @@ namespace ranges
                 return *this;
             }
         };
+
+        namespace detail
+        {
+            template<typename I>
+            struct reverse_cursor
+            {
+            private:
+                template<typename OtherI>
+                friend struct reverse_cursor;
+                I it_;
+            public:
+
+                struct mixin : basic_mixin<reverse_cursor>
+                {
+                    RANGES_CXX14_CONSTEXPR mixin() = default;
+                    RANGES_CXX14_CONSTEXPR mixin(reverse_cursor pos)
+                        : basic_mixin<reverse_cursor>{std::move(pos)}
+                    {}
+                    RANGES_CXX14_CONSTEXPR mixin(I it) : mixin(reverse_cursor{it})
+                    {}
+                    RANGES_CXX14_CONSTEXPR I base() const
+                    {
+                        return this->get().base();
+                    }
+                };
+
+                RANGES_CXX14_CONSTEXPR reverse_cursor(I it) : it_(std::move(it)) {}
+                template<typename U, CONCEPT_REQUIRES_(Convertible<U, I>{})>
+                RANGES_CXX14_CONSTEXPR reverse_cursor(reverse_cursor<U> const& u)
+                        : it_(u.base()) {}
+
+                template<typename U, CONCEPT_REQUIRES_(Convertible<I, U>{})>
+                RANGES_CXX14_CONSTEXPR operator reverse_cursor<U>() {
+                    return U{it_};
+                }
```

Comment:
You're liable to get conversion ambiguities if you have both a converting constructor and a conversion operator. Remove the conversion operator and keep the constructor.